### DISCUSSION TITLE
[advanced-typeguard] Unexpectedly pass

### DISCUSSION THREAD

**Discussion by F-park** [laike9m/Python-Type-Challenges#99] [Ideas]
The below code can pass the challenge.

```py
from typing import Any, TypeGuard


def is_string(value: Any) -> TypeGuard[str]:
    ...
```

However, If we replace `...` with `pass`, it fail

```py
from typing import Any, TypeGuard


def is_string(value: Any) -> TypeGuard[str]:
    pass
```

Shall we modify it?

**Comment by laike9m** [comment]
This is the expected behavior. pyright (and probably other type checkers) treat `...` differently than `pass`. Normally `...` will not trigger type error like you observed.

**Reply by laike9m** [reply]
Reference

https://github.com/microsoft/pyright/issues/5985#issuecomment-1728078563

**Reply by F-park** [reply]
However, the solution of `advanced-typeguard` is as below.

```py
def is_string(value: Any) -> TypeGuard[str]:
    return isinstance(value, str)
```

Just use `...` to pass the challenge is confusing.

**Comment by laike9m** [comment]
I see your point. Adding `return isinstance(value, str)` to the question file seems to be the best fix, which also makes the intent more clear. wdyt?

**Reply by F-park** [reply]
`return isinstance(value, str)` is already in the question file, I am not sure we can replace `...` with `"""Write your code here"""` to avoid this bug？

**Reply by laike9m** [reply]
I mean, we can put the following in the question file

```
def is_string(value: Any):
    return isinstance(value, str)
```

Without annotating the return value with `TypeGuard[str]`, it will also fail, which is what we want

**Reply by F-park** [reply]
Looks good to me

**Reply by laike9m** [reply]
ok I'll change it this way.

**Comment by laike9m** [comment]
Fixed in https://github.com/laike9m/Python-Type-Challenges/commit/0fa2d80e1554fe66a54ae1441723c92665a35a64
